src/bpfprogram.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "mapmanager.h"
+
+#include <cstdint>
+#include <optional>
+#include <string>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+namespace bpftrace {
+
+using BpfBytecode = std::unordered_map<std::string, std::vector<uint8_t>>;
+
+class BPFtrace;
+
+class BpfProgram
+{
+public:
+  static std::optional<BpfProgram> CreateFromBytecode(
+      const BpfBytecode &bytecode,
+      const char *name,
```

Comment:
```suggestion
      std::string_view name,
```
Just a nit and I know it wasn't you, but this is C++!

Reply:
In the end, I used `const std::string &` as the values passed are strings anyways.